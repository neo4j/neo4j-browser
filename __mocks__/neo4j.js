/*
 * Copyright (c) 2002-2018 "Neo4j, Inc"
 * Network Engine for Objects in Lund AB [http://neotechnology.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

function integerFn (val) {
  this.val = val
}
integerFn.prototype.toString = function () {
  return this.val.toString()
}

var out = {
  v1: {
    Int: integerFn,
    isInt: function (val) {
      return val instanceof integerFn
    },
    types: {
      Node: function Node (id, labels, properties) {
        this.identity = id
        this.labels = labels
        this.properties = properties
      },
      Relationship: function Relationship (id, start, end, type, properties) {
        this.identity = id
        this.start = start
        this.end = end
        this.type = type
        this.properties = properties
      },
      Path: function Path (start, end, segments) {
        this.start = start
        this.end = end
        this.segments = segments
        this.length = segments.length
      },
      PathSegment: function PathSegment (start, relationship, end) {
        this.start = start
        this.relationship = relationship
        this.end = end
      }
    },
    Integer: function Integer ({ low, high }) {
      this.low = low
      this.high = high
    }
  }
}

out.v1.types.Node.prototype.toString = function () {
  return 'node'
}
out.v1.types.Relationship.prototype.toString = function () {
  return 'rel'
}
out.v1.types.Path.prototype.toString = function () {
  return 'path'
}
out.v1.types.PathSegment.prototype.toString = function () {
  return 'pathsegment'
}
out.v1.Integer.prototype.toInt = function () {
  return this.low
}
out.v1.int = val => new out.v1.Integer(val)

module.exports = out
