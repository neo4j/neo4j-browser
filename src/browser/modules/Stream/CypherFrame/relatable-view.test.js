/*
 * Copyright (c) 2002-2020 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import { render } from '@testing-library/react'
import neo4j from 'neo4j-driver'

import {
  RelatableViewComponent as RelatableView,
  RelatableStatusbarComponent as RelatableStatusbar
} from './relatable-view'

describe('RelatableViews', () => {
  describe('RelatableView', () => {
    test('displays bodyMessage if no rows', () => {
      // Given
      const props = {
        result: {
          records: [],
          summary: {
            resultAvailableAfter: neo4j.int(5),
            resultConsumedAfter: neo4j.int(5)
          }
        }
      }

      // When
      const { container } = render(
        <RelatableView {...props} maxFieldItems={1000} maxRows={1000} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
    test('does not display bodyMessage if rows, and escapes HTML', () => {
      const value = 'String with HTML <strong>in</strong> it'
      const result = {
        records: [{ keys: ['x'], _fields: [value], get: () => value }],
        summary: {
          resultAvailableAfter: neo4j.int(5),
          resultConsumedAfter: neo4j.int(5)
        }
      }

      // When
      const { container } = render(
        <RelatableView result={result} maxFieldItems={1000} maxRows={1000} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
  })
  describe('TableStatusbar', () => {
    test('displays no statusBarMessage', () => {
      // Given
      const props = { result: {}, maxRows: 0 }

      // When
      const { container } = render(
        <RelatableStatusbar {...props} maxFieldItems={1000} maxRows={1000} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
    test('displays statusBarMessage', () => {
      // Given
      const props = {
        query: 'MATCH xx0',
        status: 'success',
        result: {
          summary: {
            resultAvailableAfter: neo4j.int(5),
            resultConsumedAfter: neo4j.int(5)
          },
          maxRows: 100,
          records: [{ res: 'xx3' }]
        }
      }

      // When
      const { container } = render(
        <RelatableStatusbar {...props} maxFieldItems={1000} maxRows={1000} />
      )

      // Then
      expect(container).toMatchSnapshot()
    })
  })
})
