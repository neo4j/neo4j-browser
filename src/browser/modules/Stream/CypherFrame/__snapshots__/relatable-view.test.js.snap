// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatableViews RelatableView displays bodyMessage if no rows 1`] = `
<div>
  <div
    class="styled__StyledStatsBar-sc-1dtvgs1-23 gWjkIU"
  >
    <div
      class="Ellipsis-sc-1dn9bou-0 gzqRPC"
    >
      (no changes, no records)
    </div>
  </div>
</div>
`;

exports[`RelatableViews RelatableView does not display bodyMessage if rows, and escapes HTML 1`] = `
<div>
  <div
    class="relatable-viewstyled__RelatableStyleWrapper-sc-4f9z7m-0 bbriTa"
  >
    <div
      class="relatable css-1dne9dv"
    >
      <table
        class="ui basic table relatable__table "
        role="table"
      >
        <thead
          class=""
        >
          <tr
            class="relatable__table-row relatable__table-header-row"
            role="row"
          >
            <th
              class="collapsing relatable__table-cell relatable__table-header-cell relatable__table-header-actions-cell"
            >
              <span
                class="relatable__table-row-actions"
              />
            </th>
            <th
              class="relatable__table-cell relatable__table-header-cell"
              colspan="1"
              role="columnheader"
            >
              x
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class="relatable__table-row relatable__table-body-row "
            role="row"
          >
            <td
              class="relatable__table-cell relatable__table-body-cell relatable__table-row-actions-cell"
            >
              <div
                class="ui label relatable__table-row-number"
              >
                1
              </div>
              <span
                class="relatable__table-row-actions"
              />
            </td>
            <td
              class="relatable__table-cell relatable__table-body-cell"
              role="cell"
            >
              <span>
                "String with HTML &lt;strong&gt;in&lt;/strong&gt; it"
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`RelatableViews TableStatusbar displays no statusBarMessage 1`] = `
<div>
  <div
    class="styled__StyledStatsBar-sc-1dtvgs1-23 gWjkIU"
  >
    <div
      class="Ellipsis-sc-1dn9bou-0 gzqRPC"
    />
  </div>
</div>
`;

exports[`RelatableViews TableStatusbar displays statusBarMessage 1`] = `
<div>
  <div
    class="styled__StyledStatsBar-sc-1dtvgs1-23 gWjkIU"
  >
    <div
      class="Ellipsis-sc-1dn9bou-0 gzqRPC"
    >
      Started streaming 1 records after 5 ms and completed after 10  ms.
    </div>
  </div>
</div>
`;
